File: Structy/Hashing/ExclusiveItems_Optimized.java
import java.util.List;
import java.util.ArrayList;
import java.util.HashSet;

class Source {
  public static List<Integer> exclusiveItems(List<Integer> a, List<Integer> b) {
    HashSet<Integer> setA = new HashSet<>(a);
    HashSet<Integer> setB = new HashSet<>(b);
    List<Integer> result = new ArrayList<>();

    for (int elem : a) {
      if(!setB.contains(elem)){
        result.add(elem);
      }
    }
    for (int elem : b) {
      if(!setA.contains(elem)){
        result.add(elem);
      }
    }
    return result;
  }

  public static void run() {
    // this function behaves as `main()` for the 'run' command
    // you may sandbox in this function , but should not remove it
  }
}

File: Structy/Python/BinarySearch/ExclusiveItems_Optimized.py
def exclusive_items(a, b):
  
  difference = []
  set_a = set(a)
  set_b = set(b)

  for item in a:
    if item not in set_b:
      difference.append(item)

  for item in b:
    if item not in set_a:
      difference.append(item)

  return difference